draft-ietf-quic-recovery.md

```diff
@@ -1001,12 +1001,21 @@ after idle periods, such as those proposed for TCP in {{?RFC7661}}.
 
 ## Discarding Packet Number Space State
 
-When keys for an packet number space are discarded, any packets sent with those
-keys are removed from the count of bytes in flight.  No loss events will occur
-any in-flight packets from that space, as a result of discarding loss recovery
-state (see {{discard-initial}}).  Note that it is expected that keys are
-discarded after those packets would be declared lost, but Initial secrets are
-destroyed earlier.
+When keys for an packet number space are discarded, any in-flight packets
+sent with those keys are removed from the count of bytes in flight.  Loss
+recovery state is also discarded, so no loss events will occur for any
+in-flight packets from that space(see {{discard-initial}}).  Note that it is
+expected that keys are discarded after those packets would be declared lost,
+but Initial secrets are destroyed earlier.
+
+## Discarding 0-RTT packet state
+
+When 0-RTT is rejected, all in-flight 0-RTT packets still are removed from
+the count of bytes in flight.  Loss recovery state is also discarded, so no
+loss events will occur for any in-flight 0-RTT packets.  If a server does not
+buffer out-of-order packets and accepts 0-RTT, there is a chance of declaring
+0-RTT packets lost that were actually discarded by the server, but this is
+expected to be infrequent.
```

Comment:
The out-of-order sentence probably needs to be its own paragraph - because it's talking about accepting 0-RTT (as opposed to rejecting it).  Not sure how that affects the flow of the section though.

Reply:
Do we need to care about loss?

I ask this based on the assumption that a server will retain the 0-RTT read key for 3 PTO after obtaining 1-RTT read key.

Reply:
The problem here is that reordering will manifest as loss, that's all.  As the text says, that's rare, and the effect is limited to having to send the data again and maybe having congestion windows docked.

Reply:
My bad. I incorrectly assumed that the text is talking about 0-RTT packets arriving late. Actually, it is talking about 0-RTT packets arriving before Initial.

I am not sure if others would have the same confusion as I had, but changing "but does not buffer out of order packets and there is reordering" to "but does not buffer 0-RTT packets arriving before Initial packets" might better clarify what is being discussed here.

Reply:
That would be an improvement, I agree.